packages/tasit-contract-based-account/src/GnosisSafe.js

```diff
@@ -11,14 +11,6 @@ const { abi: gnosisSafeABI } = GnosisSafeInfo;
 const { abi: erc20ABI } = MyERC20Full;
 const { abi: erc721ABI } = MyERC721Full;
 
-// TODO: Go deep on gas handling.
-// Without that, VM returns a revert error instead of out of gas error.
-// See: https://github.com/tasitlabs/TasitSDK/issues/173
-const gasParams = {
```

Comment:
We might want to use higher gas params for the Gnosis Safe tests than we need to for the regular `tasit-action` tests

Reply:
Yes, I'm using these (higher) params as default now but I think that we need to work on estimation. Maybe as the next issue?

Reply:
Yep, I think working on estimation can make sense for an issue pretty soon.

What I'm advocating for here is that we use these higher params as a default everywhere except for the tests in `tasit-action` itself, where we know `gasParams` can be lower. So `tasit-action` could override the default in its own tests?

Reply:
For sure, we can do that.
I've thought that will be desirable not exposing gas setup thru our API and having an excellent gas estimation solution.

Reply:
I agree that requiring gas setup thru our API is not desirable. Exposing it as optional with a default might be okay though - as long as this isn't in a very prominent, distracting place in the docs

Reply:
Ok, how about the API?
For now we're almost always using the `Action` thru the `Contract` object.
Should we expose the gas customization for each contract write contract (as ethers does):
```javascript
contract.callWrite(arg1, arg2, gasParams);
```

Or having a setter for the contract?

```javascript
contract.gasParams(gasParams);
contract.callWrite(arg1, arg2);
```

Reply:
What do you mean by `contract write contract`?

Reply:
I could imagine a contract having two write functions with very different gas needs, so allowing custom ones for each function seems to make sense (although spending time on this right now is a low priority - we can track it in an issue)

Reply:
> What do you mean by `contract write contract`?

Sorry: contract write function

Reply:
We already have an issue about that: https://github.com/tasitlabs/TasitSDK/issues/187

Reply:
> Sorry: contract write function

Ah ok, makes sense That's what I assumed in my comment, so that comment sitll stands.

Reply:
> We already have an issue about that: #187

But that issue is mostly about allowing for not specifying gas params outside the SDK

Whereas what we're discussing here is letting users optionally specify gas params outside the SDK if they want. Right? Should we mention that in that issue and broaden its scope, or create a new issue?

Reply:
I think a specific issue will be better. Extracted to: https://github.com/tasitlabs/TasitSDK/issues/371

Reply:
Sounds good - thanks

Reply:
Resolving this